### ISSUE TITLE
Hide user list from normal users

### ISSUE THREAD

**Issue by jeffbaumes** [girder/girder#2252]
See https://github.com/girder/girder/issues/2251#issuecomment-320093123 for rationale.

**Comment by brianhelba** [comment]
Is this a proposal to disable the listing (of all users by non-admin users) at the API level, or just cosmetically in the web client UI?

**Comment by zachmullen** [comment]
If it's a security issue, then it's also a security issue to allow discovery of users through the search bar, since it's trivial to find all users in the system by simply searching by string prefix.

Users are (on the back-end) proper access controlled models, we just never really exposed front-end controls or support for making users private. As such, they are all set as publicly visible, though it's easy for them to make any of their *data* private.

**Comment by brianhelba** [comment]
See #1470 for my never-completed attempt to implement private users. There are a number of bugs and corner cases that need to be resolved before we can support private / access controlled users in a truly robust way, but do note that [tightly-controlled plugins](https://github.com/ImageMarkup/isic-archive) are able to handle it right now.

**Comment by jeffbaumes** [comment]
Agreed @zachmullen. This SO question shows that if you really want to scrape all users of a site, it is probably possible, even if there is not an endpoint for listing every user explicitly:

https://stackoverflow.com/questions/7329978/how-to-list-all-github-users

So it's a fool's errand to not allow user enumeration by any means, and I'll take back my security concern.

My rationale then is limited to the (admittedly not super-strong) reasons of UI clutter for non-admins and complexity of adding more sophisticated things like user count, or batch operations / checkboxes to the Users view. If it was admin-only, we can focus on making it more useful as an admin view and not worry about making that view work for non-admins.

**Comment by brianhelba** [comment]
Non-admins may definitely want to visit other user's home directories to access files stored there. All users have a `Public` folder created for them by default, but the fact that every user has a folder of this name actually makes it very difficult to use the search widget to access it within a particular user. Searching for the name of a user will also pull you directly to their home directory, but for a user with a common or ambiguous name, it may be much easier to determine the intended one via the user listing page (since the search box clears after every use).

**Comment by gdolle** [comment]
I think also that this is a security issue. 
At least we should have the possibility to hide the user login in the user list (in particular when we are using LDAP for several services).
